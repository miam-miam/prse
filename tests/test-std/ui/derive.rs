#![allow(unused)]
use prse::{parse, LendingFromStr};

#[derive(LendingFromStr)]
#[prse = "{x} - {y}"]
union A {
    x: usize,
    y: u32,
}

#[derive(LendingFromStr)]
struct B {
    #[prse = "{x} - {y}"]
    x: usize,
    y: usize,
}

#[derive(LendingFromStr)]
struct C {
    y: usize,
    #[prse = "{x} - {y}"]
    x: usize,
}

#[derive(LendingFromStr)]
#[prse = "D: {}"]
struct D;

#[derive(LendingFromStr)]
#[prse = "E: {0}"]
struct E;

#[derive(LendingFromStr)]
#[prse = "F: {x}"]
struct F;

#[derive(LendingFromStr)]
#[prse = "F: {x} {y}"]
struct G {
    y: usize,
    #[prse = "{x} - {y}"]
    x: usize,
}

#[derive(LendingFromStr)]
#[prse = "G: {a}"]
#[prse = "G: {a} "]
struct H {
    a: usize,
}

#[derive(LendingFromStr)]
#[prse = "I"]
enum I {
    Unit,
}

#[derive(LendingFromStr)]
#[prse = "J"]
enum J {
    #[prse = "J"]
    Unit,
}

#[derive(LendingFromStr)]
enum K {
    Tup(#[prse = "K: {}"] usize, usize),
}

#[derive(LendingFromStr)]
enum L {
    #[prse = "L"]
    Unit,
    #[prse = "L1"]
    Unit2,
    Unit3,
}

#[derive(LendingFromStr)]
enum M {
    S {
        x: usize,
        #[prse = "Test: {y}"]
        y: usize,
    },
}

#[derive(LendingFromStr)]
enum N {
    #[prse("N")]
    N,
}

#[derive(LendingFromStr)]
enum O {
    #[prse]
    O,
}

#[derive(LendingFromStr)]
enum P {
    #[prse = "{:,:0}"]
    P1(u32),
}

#[derive(LendingFromStr)]
#[prse = "{x:,:0}"]
struct Q {
    x: u32,
}

fn main() {}
