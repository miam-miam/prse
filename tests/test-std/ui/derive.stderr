error: The derive macro does not currently support unions.
 --> ui/derive.rs:6:1
  |
6 | union A {
  | ^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:13:6
   |
13 |     #[prse = "{x} - {y}"]
   |      ^^^^^^^^^^^^^^^^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:21:6
   |
21 |     #[prse = "{x} - {y}"]
   |      ^^^^^^^^^^^^^^^^^^^^

error: A unit field cannot contain variables
  --> ui/derive.rs:26:10
   |
26 | #[prse = "D: {}"]
   |          ^^^^^^^

error: A unit field cannot contain variables
  --> ui/derive.rs:30:10
   |
30 | #[prse = "E: {0}"]
   |          ^^^^^^^^

error: A unit field cannot contain variables
  --> ui/derive.rs:34:10
   |
34 | #[prse = "F: {x}"]
   |          ^^^^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:41:6
   |
41 |     #[prse = "{x} - {y}"]
   |      ^^^^^^^^^^^^^^^^^^^^

error: Expected only a single prse attribute.
  --> ui/derive.rs:47:2
   |
47 | #[prse = "G: {a} "]
   |  ^^^^^^^^^^^^^^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:53:2
   |
53 | #[prse = "I"]
   |  ^^^^^^^^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:59:2
   |
59 | #[prse = "J"]
   |  ^^^^^^^^^^^^

error: Unexpected prse attribute.
  --> ui/derive.rs:67:10
   |
67 |     Tup(#[prse = "K: {}"] usize, usize),
   |          ^^^^^^^^^^^^^^^^

error: The derive macro must either have an attribute on each field or none at all.
  --> ui/derive.rs:71:6
   |
71 | enum L {
   |      ^

error: Unexpected prse attribute.
  --> ui/derive.rs:83:10
   |
83 |         #[prse = "Test: {y}"]
   |          ^^^^^^^^^^^^^^^^^^^^

error: prse attribute must be of the form #[prse = "parse_string"]
  --> ui/derive.rs:90:6
   |
90 |     #[prse("N")]
   |      ^^^^^^^^^^^

error: prse attribute must be of the form #[prse = "parse_string"]
  --> ui/derive.rs:96:6
   |
96 |     #[prse]
   |      ^^^^^^
